src/msg/Messenger.cc

```diff
@@ -21,7 +21,7 @@ Messenger *Messenger::create_client_messenger(CephContext *cct, std::string lnam
 uint64_t Messenger::get_pid_nonce()
 {
   uint64_t nonce = getpid();
-  if (nonce == 1) {
+  if (nonce <= 10 || getenv("CEPH_CONTAINER_HAS_INIT")) {
```

Comment:
@liewegas Mind changing "pid 2" to "pid 10" in the commit title and commit message, too?